ote/src/clj/ote/services/external.clj

```diff
@@ -20,25 +33,46 @@
 (defn parse-response->csv
   "Convert given vector to map where map key is given in the first line of csv file."
   [csv-data]
-  (map zipmap
-       (->> (first csv-data) ;; First row is the header
-            (map keyword) ;; Drop if you want string keys instead
-            repeat)
-       (rest csv-data)))
+  (let [headers (first csv-data)
+        valid-header? (valid-csv-header? headers)
+        parsed-data (when valid-header?
+                      (map (fn [[business-id name]]
+                             {::t-service/business-id business-id
+                              ::t-service/name name})
+                            (rest csv-data)))
+        validated-data (filter #(valid-business-id? (::t-service/business-id %)) parsed-data)
+        filtered-data  (map #(dissoc % :remove) validated-data )]
+    {:result filtered-data
+     :failed-count (- (count parsed-data) (count filtered-data))}))
 
-(defn- check-csv
+(defn save-companies
+  "Save business-ids, company names to db"
+  [db data]
+   (let [data (if (nil? (::t-service/id data))
```

Comment:
We have `ote.db.modification` ns and the `created` and `modified` columns should be mapped to keywords in that namespace as is done with other tables.

Also use `with-modification-fields` helper that conditionally adds creation/modification.

Reply:
I thought of using with-modification-fields function. But in it's current form it also requires to table to have a user. service-company table is mainly used with our scheduler so I'm not adding user-id to it.
So I thought that there is no reason to map those fields any differently as any other field in that table.